test/integration/master_routes_test.go

```diff
@@ -183,17 +188,79 @@ func TestRootRedirect(t *testing.T) {
 		t.Fatalf("Unexpected index: \ngot=%v,\n\n expected=%v,\n\ndiff=%v", got.Paths, expectedIndex, diff.ObjectDiff(expectedIndex, got.Paths))
 	}
 
-	req, err = http.NewRequest("GET", masterConfig.OAuthConfig.MasterPublicURL, nil)
-	req.Header.Set("Accept", "text/html")
-	resp, err = transport.RoundTrip(req)
+	// Create fake config map to test console redirect
+	webconsoleConfigScheme := runtime.NewScheme()
+	webconsoleConfigCodecs := serializer.NewCodecFactory(webconsoleConfigScheme)
+	webConsoleURL := "https://127.0.0.42/console"
+
+	if err := webconsoleconfigv1.AddToScheme(webconsoleConfigScheme); err != nil {
+		t.Fatalf("Unextecpted error: %v", err)
+	}
+
+	clusterAdminKubeClientset, err := testutil.GetClusterAdminKubeClient(clusterAdminKubeConfig)
+	if err != nil {
+		t.Fatalf("Unextecpted error: %v", err)
+	}
+
+	ns := kapi.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "openshift-web-console",
+		},
+	}
+	if _, err = clusterAdminKubeClientset.Core().Namespaces().Create(&ns); err != nil {
+		t.Fatalf("Unextecpted error: %v", err)
+	}
+
+	consoleConfig := webconsoleconfigv1.WebConsoleConfiguration{
+		ClusterInfo: webconsoleconfigv1.ClusterInfo{
+			ConsolePublicURL: webConsoleURL,
+		},
+	}
+	data, err := runtime.Encode(webconsoleConfigCodecs.LegacyCodec(webconsoleconfigv1.SchemeGroupVersion), &consoleConfig)
 	if err != nil {
-		t.Errorf("Unexpected error: %v", err)
+		t.Fatalf("Unextecpted error: %v", err)
+	}
+	configMap := kapi.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift-web-console",
+			Name:      "webconsole-config",
+		},
+		Data: map[string]string{
+			"webconsole-config.yaml": string(data),
+		},
+	}
+	if _, err = clusterAdminKubeClientset.Core().ConfigMaps("openshift-web-console").Create(&configMap); err != nil {
+		t.Fatalf("Unextecpted error: %v", err)
+	}
+
+	// try three times then give up
```

Comment:
A little weird given the live lookup.  I guess if we ever change it back.  Both method had merit.

Reply:
I had issues getting this to work, and at some point I got it spin forever w/o any timeout, so I decided it was a better idea to limit it and terminate early